CHANGELOG.md

```diff
@@ -1,3 +1,227 @@
+  - `writeCheckpoint` option has been removed. Use `abortProjection` if wishing to set this to `false`.
+- `abortProjection`:
+  - `abortProjection` method has been added.
+- `deleteProjection`:
+  - `deleteEmittedStreams` option now defaults to `false`.
+  - `deleteStateStream` option now defaults to `false`.
+  - `deleteCheckpointStream` option now defaults to `false`.
+
+### Features
+
+#### Persistent Subscriptions to all
+
+The client now supports persistent subscriptions to the all stream, for server version `21.10` and greater.
+
+- Added `createPersistentSubscriptionToAll` method. [View](https://github.com/EventStore/EventStore-Client-NodeJS/commit/f7d5f7db012000ef82dac0e8b6fbcd4d598d812a)
+
+```ts
+const groupName = "my_persistent_subscription";
+const settings = persistentSubscriptionToAllSettingsFromDefaults({
+  // start from the beginning of the all stream
+  startFrom: START,
+});
+// Filter only events with type matching regex
+const filter = eventTypeFilter({
+  regex: "^[0-9]*_regex_filter_eventType_[A-z]*$",
+});
+
+// Create a persistent subscrition to the all stream
+await client.createPersistentSubscriptionToAll(groupName, settings, { filter });
+```
+
+- Added `subscribeToPersistentSubscriptionToAll` method. [View](https://github.com/EventStore/EventStore-Client-NodeJS/commit/bf660ef1d82c02cf577de439db2c5b5529399c0a)
+
+```ts
+// Connect to the created persistent subscription to all stream
+const psToAll = client.subscribeToPersistentSubscriptionToAll(groupName);
+
+for await (const event of psToAll) {
+  doSomethingWithEvent(event);
+  // Acknowledge that the event has been handled
+  await psToAll.ack(event);
+}
+```
+
+- Added `updatePersistentSubscriptionToAll` method. [View](https://github.com/EventStore/EventStore-Client-NodeJS/commit/75aa261f6307408383b20a98447782ca77ce75c7)
+
+```ts
+const updatedSettings =  persistentSubscriptionToAllSettingsFromDefaults({
+  // Ensure that our previous settings are persisted
+  ...settings,
+  // Enable extra statistics
+  extraStatistics: true,
+});
+
+// Update the persistent subscription to use new settings.
+await client.updatePersistentSubscriptionToAll(groupName, updatedSettings)l
+
+```
+
+- Added `deletePersistentSubscriptionToAll` method. [View](https://github.com/EventStore/EventStore-Client-NodeJS/commit/cf4fa765282fe00cc3195acafcb75421fcd5f733)
+
+```ts
+// delete the unwanted persistent subscription to all
+await client.deletePersistentSubscriptionToAll(groupName);
+```
+
+#### Other features
+
+- When connected to server version `21.10` and greater, the client will now send appended events over a single duplex stream, resulting in much faster append times. [View](https://github.com/EventStore/EventStore-Client-NodeJS/commit/4e764c1d3ca20627553ad16588a259d63283a31b)
+- Client now internally checks if the server supports an api before sending a request. [View](https://github.com/EventStore/EventStore-Client-NodeJS/commit/4122f6f17756c7a4deff0288bfb8af9adfca4aee)
```

Comment:
What's the 'error' returned if a given API is not supported on the server ?

Reply:
An `UnsupportedError`, with a message like: `createPersistentSubscriptionToAll requires server version 21.10 or higher.`

https://github.com/EventStore/EventStore-Client-NodeJS/blob/master/src/__test__/persistentSubscription/createPersistentSubscriptionToAll.test.ts#L40

or it will use the older append api, in the case of `appendToStream`